lib/src/cpp/domain_participant/DomainParticipant.cpp

```
@@ -20,314 +20,437 @@
 }
 
 void clear_prefix(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_default_profile(
         const std::string& xml_file,
         const std::string& profile_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_domain_id(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& domain_id)
 {
-    throw Unsupported("Unsupported");
+    throw eprosima::qosprof::Unsupported("Unsupported");
 }
 
 void set_name(
         const std::string& xml_file,
         const std::string& profile_id,
         const std::string& name)
 {
-    throw Unsupported("Unsupported");
+    // Xerces objects to manage XML elements
+    xercesc::DOMConfiguration* config = NULL;
+    xercesc::DOMDocument* doc = NULL;
+    ParseXMLErrorHandler* error_handler = NULL;
+    xercesc::DOMImplementation* implementation = NULL;
+    xercesc::DOMLSOutput* output = NULL;
+    xercesc::XercesDOMParser* parser = NULL;
+    xercesc::DOMLSSerializer* serializer = NULL;
+    xercesc::XMLFormatTarget* target = NULL;
+
+    // XML nodes and values
+    xercesc::DOMNode* profiles_node = NULL;
+    xercesc::DOMNode* participant_node = NULL;
+    xercesc::DOMNode* rtps_node = NULL;
+    xercesc::DOMNode* name_node = NULL;
+    xercesc::DOMText* name_value = NULL;
+
+    // Obtain absolute path of the given file
+    std::string absolute_xml_file = "/home/jesus/Desktop/test.xml"; // TODO update
+
+    // Try to initialize XML workspace
+    try
+    {
+        xercesc::XMLPlatformUtils::Initialize();
+    }
+    catch (const xercesc::XMLException& toCatch)
+    {
+        // Unable to initialize XML workspace
+        throw eprosima::qosprof::FileNotFound(xercesc::XMLString::transcode(toCatch.getMessage()));
+        return;
```

Other place where probably we should throw an Error exception instead of one that we have.